plugins/org.jkiss.dbeaver.ext.test/src/org/jkiss/dbeaver/ext/test/handlers/HandlerConnectionStressTest.java

```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.test.handlers;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.jkiss.dbeaver.model.DBPDataSource;
+import org.jkiss.dbeaver.model.exec.*;
+import org.jkiss.dbeaver.model.navigator.DBNDatabaseNode;
+import org.jkiss.dbeaver.model.runtime.VoidProgressMonitor;
+import org.jkiss.dbeaver.ui.UIUtils;
+import org.jkiss.dbeaver.ui.dialogs.EnterNameDialog;
+import org.jkiss.utils.CommonUtils;
+
+import java.util.ArrayList;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.IntStream;
+
+public class HandlerConnectionStressTest extends AbstractHandler {
+    private static final int CONNECTIONS_COUNT = 50;
+    private static final int COLUMNS_COUNT = 10;
+    private static final String DUMMY_SQL =
+        """
+            with recursive all_rows as (
+                select *
+                from (values({x1},{x1}),({x2},{x2}),({x3},{x3}),({x4},{x4}),({x5},{x5})) as t_name(id, quantity)
+            ),
+            sub_select AS (
+                select id as rows_id
+                from all_rows AS data1
+                UNION ALL\s
+                select id as sub_id
+                from sub_select
+                JOIN all_rows on 1=1
+            )
+             \s
+            SELECT rows_id,rows_id,rows_id,rows_id,rows_id,rows_id,rows_id,rows_id,rows_id,rows_id
+            FROM sub_select\s
+            limit {max};
+            """;
+
+    @Override
+    public Object execute(ExecutionEvent event) throws ExecutionException {
+        String connectionsCount = EnterNameDialog.chooseName(UIUtils.getActiveWorkbenchShell(),
+            "Number of connections",
+            "");
+        if (CommonUtils.isEmpty(connectionsCount)) {
+            connectionsCount = String.valueOf(CONNECTIONS_COUNT);
+        }
+        int maxConnections = Integer.parseInt(connectionsCount);
+
+        final ISelection selection = HandlerUtil.getCurrentSelection(event);
+        if (selection instanceof IStructuredSelection) {
+            final Object element = ((IStructuredSelection) selection).getFirstElement();
+            if (element instanceof DBNDatabaseNode) {
+                stressTest(((DBNDatabaseNode) element).getDataSource(), maxConnections);
+            }
+        }
+        return null;
+    }
+
+    private void stressTest(DBPDataSource dataSource, int maxConnections) {
+        var contexts = new ArrayList<DBCExecutionContext>();
+        try {
+            IntStream.range(0, maxConnections).parallel()
+                .forEach(i -> {
+                        try {
+                            contexts.add(dataSource.getDefaultInstance()
```

Comment:
Since it's `parallel` and the collection is not thread-safe, it might result in race conditions. I'd recommend doing all steps inside a single parallel stream instead of introducing an intermediate step.